Feature/wave api rdkb adaptations

morantr opened the pull request:
Add adaptations to compile prplmesh on RDKB tree.

tomereli commented on controller/src/beerocks/bml/bml.cpp:
```diff
@@ -556,3 +556,77 @@ int bml_channel_selection(BML_CTX ctx, const char *al_mac, const char *ruid)
+        return (-BML_RET_INVALID_ARGS);
+	}
+
+	// TODO: call suit bml api
```
call suitable (nitpick)

tomereli commented on controller/src/beerocks/bml/bml_defs.h:
```diff
@@ -549,6 +553,45 @@ struct BML_STATS_ITER {
+struct BML_DCS_NEIGHBOR_AP {
+    //char  ap_Radio[64];
+    //The Radio that detected the neighboring WiFi SSID.
+    char ap_SSID[64];
```
Please use static const definitions instead of hard coded numbers

tomereli replied:
This should also be changed now cause I doubt that @itayx already changed it in his branches.

itayx replied:
Actually I did

tomereli replied:
So OK :)

tomereli commented on controller/src/beerocks/bml/bml.cpp:
```diff
@@ -556,3 +556,77 @@ int bml_channel_selection(BML_CTX ctx, const char *al_mac, const char *ruid)
+	}
+
+	// TODO: call suit bml api
+	return 0;
```
I think that we should return failure until these stubs are implemented. This goes to all the stubs below and above this one. and also add LOG.

tomereli replied:
This MUST return failure, I don't care if @itayx will change it soon - cause until he does, the API should not report success when it's not implemented..

tomereli reviewed (changes requested):
Last commit can't be merged - it was done by @arnout  and @rmelotte for the CI, we need to find a different solution.

morantr commented:
@tomereli There is no point changing what I did. Itay should change it very soon.

tomereli commented:
> @tomereli There is no point changing what I did. Itay should change it very soon.

Not sure which comment you mean but the changes requested are mainly due to the last commit. @rmelotte added the soft link and **this is why your PR failed CI.**

rmelotte commented:
> Not sure which comment you mean but the changes requested are mainly due to the last commit. @rmelotte added the soft link and this is probably why your PR failed CI.

The soft-link was there to make it possible to use any docker image regardless of it's default entrypoint and not break the builds for anyone.
The plan was to remove it after the required changes made their way to all the development branches (it should be the case now).
The reason it's still failing is because the images on dockerhub haven't been updated.
However I think it would be better to just override the entrypoint whenever we run the image (from the build script for instance).
Also if we decided to update the images on dockerhub at some point, we might just as well remove the entrypoint entirely, since it makes things easier for the CI.

I'll make a patch and see if I didn't miss something.

morantr committed:
tools: Add wav-api path to rdkb workspace

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

morantr committed:
bml: Add dcs related stubs to compile rdkb successfully

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

morantr committed:
tools: Match RDKB workspace to prplmesh workspace

Copy from prplmesh workspace file diffrences into rdkb workspace file  so it
will be match.

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

morantr committed:
Remove root dir softlink

RDKB fail to compile prplmesh with rootdir softlink inside because it tries to
get to the lower level but fails since it has too many levels.
Remove the softlink to prevent it.

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

rmelotte committed:
tools: docker: override the entrypoint from the build script

Commit cf40f6a651ad58235b31c866c2a8bd6b70b67215 changed the default
build location to ./build instead of ../build. The images on dockerhub
weren't updated to not break the builds for branches relying on the
old build location. Instead, a temporary 'prplMesh -> .' soft-link was
added at the root of the repository until the changes made their way
to all the development branches.

Now all the branches should have the required changes, so we can
remove the soft-link.

In the meantime the images on dockerhub weren't updated, so the build
will still fail for Travis and for anyone who didn't rebuild the image
locally.

Since Gitlab-CI anyway uses an interactive entrypoint, we might not
want to update the dockerhub images to use "tools/maptools.py" but to
plainly remove the entrypoint.

To make it possible to use any image regardless of what it's
entrypoint is, override the entrypoint from the build script. Since
the default entrypoint added "build" as an argument, add it when
calling docker run.

Signed-off-by: Raphaël Mélotte <raphael.melotte@mind.be>

rmelotte commented:
The last 2 commits were intended to be squashed together because removing the soft-link without any other changes won't work (unless the images on dockerhub are updated to reflect this specific change, but I don't think it's worth it to do that).
